grpc/naming.go

```diff
@@ -84,10 +103,15 @@ func (w *watcher) Next() ([]*naming.Update, error) {
 	}
 
 	var lastErr error
-	for i := 0; i < MaximumConsecutiveErrors; i++ {
+	consecutiveErrors := 0
+	for {
 		freshTargets, err := w.resolver.Lookup(w.domainName)
 		if err != nil {
 			lastErr = err
+			consecutiveErrors += 1
+			if consecutiveErrors >= w.maxConsecutiveErrors && w.maxConsecutiveErrors != NoRetryLimit {
```

Comment:
This line might read nicer if you swap the two conditions as we want to make clear that if NoRetryLimit is specified then nothing else is worth comparing.

```
w.maxConsecutiveErrors != NoRetryLimit && onsecutiveErrors >= w.maxConsecutiveErrors
```